### PR TITLE
Add the addShould function to ProxyQuery

### PR DESCRIPTION
**Pull request by chervaliery**
The addShould function was missing

### CODE REVIEW

**Comment by soullivaneuh** [comment]
@greg0ire only the concerned file is breaking CS: https://travis-ci.org/sonata-project/SonataAdminSearchBundle/jobs/68136202#L139

@chervaliery Just run `make cs` to fix it.

**Comment by chervaliery** [comment]
I fixed the CS

File: ProxyQuery/ElasticaProxyQuery.php [hunk 1]
```diff
@@ -189,6 +189,12 @@ public function addMustNot($args)
         $this->query = new \Elastica\Query($this->boolQuery);
     }
 
+    public function addShould($args)
```
Comment (soullivaneuh) [inline ProxyQuery/ElasticaProxyQuery.php:199]: Missing PHP doc block.

**Comment by chervaliery** [comment]
I had the doc block but I don't understand why the build faild

**Comment by greg0ire** [comment]
Some authentication problem… I restarted the job.

**Comment by soullivaneuh** [comment]
@greg0ire Yes it's a know problem.

Composer should go back to source if dist fail, but sometimes it doesn't.

Related issue: https://github.com/composer/composer/issues/4147

**Comment by greg0ire** [comment]
… same result : 

> [Composer\Downloader\TransportException]   
> 
>  Could not authenticate against github.com  

@Soullivaneuh : IIRC, you are aware of this problem on another project.

File: ProxyQuery/ElasticaProxyQuery.php [hunk 2]
```diff
@@ -190,6 +190,15 @@ public function addMustNot($args)
     }
 
     /**
+     * Add should part to query.
+     */
```
Comment (soullivaneuh) [inline ProxyQuery/ElasticaProxyQuery.php:198]: Missing `@param`

**Comment by soullivaneuh** [comment]
@greg0ire Try re-launching master job to refresh vendor cache before this job.

**Comment by greg0ire** [comment]
[master build](https://travis-ci.org/sonata-project/SonataAdminSearchBundle/builds/67646465) re-launched

**Comment by greg0ire** [comment]
master build alive and well

**Comment by greg0ire** [comment]
Still fails on step 9. Reloading to see if if fails always on the same step.

**Comment by greg0ire** [comment]
Failing at step9 too… very strange.

**Comment by soullivaneuh** [comment]
Indeed. This should not... :/

**Comment by greg0ire** [comment]
@chervaliery could you try temporarily removing

> composer config -q github-oauth.github.com $GITHUB_OAUTH_TOKEN

from `.travis.yml` (in a separate commit, ofc) ?

**Comment by greg0ire** [comment]
Same result, no luck…

**Comment by greg0ire** [comment]
@Soullivaneuh : what is the use of this environment variable ? In which case is it not empty ? If it is not empty, doesn't it mean we make an api key public ?

**Comment by soullivaneuh** [comment]
> what is the use of this environment variable ?

To get GitHub API auth with composer and increase download rate limit.

> In which case is it not empty ?

All excepted Pull Request. This is why we get this "error" (should fallback, not fail).

> If it is not empty, doesn't it mean we make an api key public ?

Nope, Travis is smart, Travis hide the key. :-)

See: https://travis-ci.org/sonata-project/SonataAdminSearchBundle/jobs/67646469#L82

**Comment by greg0ire** [comment]
Indeed… I guess the key is configured in the github account for the sonata-project organization, isn't it ? So for it to be used in a P.R., we would have to somehow vouch for @chervaliery … or maybe @chervaliery needs to configure something in his account so that **his** token is used when making any P.R. ?

**Comment by soullivaneuh** [comment]
It's unfortunately not possible to do that kind of thing.

Maybe you can open an issue on Travis to suggest this option! ;-)

**Comment by greg0ire** [comment]
The "ask a question" button on travis leads to SO, so [here you go](http://stackoverflow.com/questions/31030578/using-a-github-oauth-token-for-prs)

**Comment by greg0ire** [comment]
@chervaliery : I don't want to block this, so I'm going to push your branch on the repo, and if it the build passes, merge it.

**Comment by greg0ire** [comment]
See #19 , thanks @chervaliery

**Pull request closed without merging**